sechub-sereco/build.gradle

```diff
@@ -7,12 +7,17 @@
  */
  
 dependencies {
-	compile library.jsoup
+	implementation library.jsoup
+	implementation(library.dom4j) {
```

Comment:
Do we need the dom4j library because of jsoup?
We normally do all JSON parts with Jackson library...

I am not so happy having jsoup and also dom4j here again. Both are good frameworks, but less is more ... at least after log4shell etc...

If jsoup is the reason why we need dom4j we should think about usage and maybe if it could be possible to drop this dependency /replace it ?

What do you think?

Reply:
Dom4j has noting to do with jsoup. Jsoup parses HTML and is only required for the [NetsparkerHtmlToAsciiDocConverter](https://github.com/Daimler/sechub/blob/develop/sechub-sereco/src/main/java/com/daimler/sechub/sereco/importer/NetsparkerHtmlToAsciiDocConverter.java). The moment we remove the Netsparker code in our repository, Jsoup will not be required anymore.

Dom4j is a very important part of  parsing the XML reports in the SERECO importer. Until now, Dom4J was a dependency of Hibernate and therefore available. That is probably the reason why the report parsing in SERECO uses Dom4J. 

In Hibernate 5.6 the Dom4J dependency got removed and therefore the report parsing did not work in several places and it resulted in the project not being able to compile. SERECO importers using Dom4j: 

- [NessusV1XMLImporter](https://github.com/Daimler/sechub/blob/develop/sechub-sereco/src/main/java/com/daimler/sechub/sereco/importer/NessusV1XMLImporter.java#L8)
- [NetsparkerV1XMLImporter](https://github.com/Daimler/sechub/blob/develop/sechub-sereco/src/main/java/com/daimler/sechub/sereco/importer/NetsparkerV1XMLImporter.java#L7)
- [CheckmarxV1XMLImporter](https://github.com/Daimler/sechub/blob/develop/sechub-sereco/src/main/java/com/daimler/sechub/sereco/importer/CheckmarxV1XMLImporter.java#L8)

Because of those three importers it was necessary to add Dom4j as a dependency. However, Dom4J comes with the pull-parser and xpp3, which caused problems with Logback. Logback needs a XML parser to read it's own configuration files: 

- [SecHub-Server: logback-spring.xml](https://github.com/Daimler/sechub/blob/develop/sechub-server/src/main/resources/logback-spring.xml)
- [SecHub-PDS: logback-spring.xml](https://github.com/Daimler/sechub/blob/develop/sechub-pds/src/main/resources/logback-spring.xml)

Both, the pull-parser and xpp3 caused problems and therefore they need to be excluded to get the old behavior back. The old behavior relied on JAXP and the default XML parser in Java SE.

Reply:
Funny that I did not remember that I used dom4j for the Netsparker Importer  when I wrote the initial implementation... But was a long time ago ... Thanx for clarification.